Packages/io.chainsafe.web3-unity/Runtime/Scripts/Connection/ServiceAdapters/MarketplaceServiceAdapter.cs

```diff
@@ -7,14 +10,33 @@ namespace ChainSafe.Gaming.UnityPackage.Connection
     /// <summary>
     /// Enables usage of Marketplace when attached.
     /// </summary>
-    public class MarketplaceServiceAdapter : MonoBehaviour, IServiceAdapter
+    public class MarketplaceServiceAdapter : MonoBehaviour, ILightWeightServiceAdapter
     {
+        [SerializeField] private MarketplaceConfigUnity marketplaceConfig;
         public Web3Builder ConfigureServices(Web3Builder web3Builder)
         {
             return web3Builder.Configure(services =>
             {
-                services.UseMarketplace();
+                if (string.IsNullOrEmpty(marketplaceConfig.MarketplaceContractAddress))
+                    throw new InvalidDataException("Please provide an appropriate marketplace address");
+
+                if (string.IsNullOrEmpty(marketplaceConfig.MarketplaceId))
+                    throw new InvalidDataException("Please provide an appropriate marketplace id");
+
+                services.UseMarketplace(marketplaceConfig);
             });
         }
     }
+
+    [Serializable]
+    public class MarketplaceConfigUnity : IMarketplaceConfig
+    {
+        [field: SerializeField] public string EndpointOverride { get; set; }
+        [field: SerializeField] public string MarketplaceContractAbi { get; set; }
```

Comment:
This can trick people into thinking they need to provide ABI for their marketplace while it's actually being loaded by default. One possible way to fix is adding "Override" prefix to the unity field name.

Reply:
Fixed